src/client/src/components/pages/courses/courseDetails/index.jsx

```diff
@@ -17,6 +17,45 @@ class index extends Component {
     this.setState({ [name]: value });
   };
 
+  updateCourse = async obj => {
+    const confirm = await swal({
+      type: "warning",
+      html: "Are you sure that you want to update this data ?",
+      showCancelButton: true,
+      focusConfirm: false,
+      confirmButtonText: '<i class="fa fa-thumbs-up"></i> Yes',
+      confirmButtonAriaLabel: "Thumbs up",
+      cancelButtonText: '<i class="fa fa-thumbs-down"></i> No ',
+      cancelButtonAriaLabel: "Thumbs down"
+    });
+    if (confirm.value) {
+      const { id } = this.props.match.params;
+      const result = await axios(`/api/v2/course/${id}`, {
+        method: "PUT",
+        data: {
+          courseData: obj
```

Comment:
I think this update code might have a similar issue (with `course_start` and `course_end`, here) that I mentioned in #186, so I'll copy that comment here:

I think there might be a bug with the course_start and course_end fields being sent back to the server in an unexpected format:
In getDetails(), you parse course_start and course_end to get the year (so for example, the server gives you '1985-12-10T00:00:00Z" and you parse/split that to put '1985-12-10' into this.state.course_start).
Then, you take this.state and PUT that to the endpoint, which means that you're PUTting that course_start that you formatted into '1985-12-10', when the server probably needs the original format, '1985-12-10T00:00:00Z'.

I've done this bug a lot too, where I take server data, make small changes to it for displaying to the user, and then accidentally send the reformatted data, not the original format of the data, back to the server when saving it. The fix is to always try to store the raw server data in this.state, and do that date formatting inside render() instead of inside getDetails().

Reply:
I will try to do it now
thanks for your great notes @tomduggan85

Reply:
@tomduggan85 
I think the database itself is the responsible for creating the timezone
so the server dosen't need the original form
The server just need <code>1985-12-10</code> and the time zone will be created automatically
- this is a screenshot for date before updating
![alt](https://image.ibb.co/i2qQ9q/Screenshot-from-2018-11-20-21-12-48.png)
- and here after update
![alt](https://image.ibb.co/jyG42A/Screenshot-from-2018-11-20-21-15-20.png)
another note: the split function, we used it just for view the date in a readable way
What do you think @tomduggan85

Reply:
I think then for this case, it is probably fine, since you're only really using the date part of that datetime.  If the actual time was important, for example if you wanted to know the exact time of birth (for a hospital app 😀) then it would be a more serious bug that the timestamp part of the datetime is being lost when you do the PUT.

I get what you're saying about the split function, and I think in general when you do something to your data to make it more readable, you should do that data transformation in render(), or right as you're about to display the data, instead of storing it in state in its altered / readable form.

A good example of how this sometimes breaks things is when displaying money / prices.  For some product, the server GET might return `price: 7.5` and then your UI will change that to show `$7.50`.  But if you stored `$7.50` in this.state instead of `7.5` and then PUT this.state back to the server, which is expecting a numeric value, then things break 😀

Reply:
Thanks a lot I will take your great advises into account.